### PR TITLE
build.d: Dont discard -debug for BUILD=debug

### PR DESCRIPTION
**Pull request by MoonlightSentinel**

### CODE REVIEW

File: src/build.d [hunk 1]
```diff
@@ -170,6 +170,8 @@ Command-line parameters
         log("================================================================================");
         foreach (key, value; env)
             log("%s=%s", key, value);
+         foreach (key, value; flags)
```
Comment (thewilsonator) [inline src/build.d]: this has one extra space of indentation

File: src/build.d [hunk 2]
```diff
@@ -1427,17 +1429,22 @@ void args2Environment(ref string[] args)
...
     default_ = fallback value if the key doesn't exist in the global environment
+
+Returns: the value associated to key
 */
 auto getDefault(ref string[string] env, string key, string default_)
 {
+    if (auto ex = key in env)
+        return *ex;
+
     if (key in environment)
```
Comment (Geod24) [inline src/build.d]: This branch can never be taken now

Comment (MoonlightSentinel) [reply src/build.d]: Why, it's `environment`not `env`?

File: src/build.d (hunk 1 above)
Comment (MoonlightSentinel) [reply src/build.d]: FIxed

**Comment by Geod24** [comment]
/me slaps @dlang-bot around a bit with a large trout

**Pull request merged**